src/librustc_trait_selection/traits/select/mod.rs

```diff
@@ -901,10 +901,7 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                     trait_ref, result,
                 );
                 // This may overwrite the cache with the same value
-                // FIXME: Due to #50507 this overwrites the different values
-                // This should be changed to use HashMapExt::insert_same
-                // when that is fixed
-                self.tcx().evaluation_cache.insert(param_env.and(trait_ref), dep_node, result);
+                self.tcx().evaluation_cache.insert_same(param_env.and(trait_ref), dep_node, result);
```

Comment:
So, #50507 is not yet closed -- but I think it is maybe specific to parallel execution? i.e., what prompted this PR?

Reply:
I was looking for issues to work on in the list at https://github.com/rust-lang/rust/issues/48685, and found a reference to that issue. I began investigating the `panic` mentioned in the issue description, and was unable to reproduce it.